compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -1158,6 +1158,19 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                 );
             }
 
+            if path_segment.ident.name == sym::ok_or && is_diagnostic_item(sym::Option, next_ty) {
+                err.span_suggestion(
+                    path_segment.ident.span,
+                    format!(
+                        "`?` expected `{}` for `Err` variant but found `{:?}`. Did you meant to use `ok_or_else`?",
```

Comment:
Per `rustc` [Suggestion style guide](https://rustc-dev-guide.rust-lang.org/diagnostics.html?highlight=diagnostics#suggestion-style-guide):
```suggestion
                        "`?` expected `{}` for `Err` variant but found `{:?}`. Use the `ok_or_else` method to pass a closure",
```